p2p/swarm.go

```diff
@@ -403,7 +401,8 @@ Loop:
 		case con := <-closing:
 			go s.retryOrReplace(con.RemotePublicKey()) //todo notify dht?
 		case nce := <-newConnEvents:
-			go func(nce net.NewConnectionEvent) { s.dht.Update(nce.Node); s.addIncomingPeer(nce.Node.PublicKey()) }(nce)
```

Comment:
As we talked, think if you really need the "go" call inside. Generally, prefer buffering over go calls.

Reply:
both send on buffered chans so should be ok.